lib/veil/cipher/v2.rb

```diff
@@ -0,0 +1,41 @@
+require "base64"
+require "openssl"
+
+module Veil
+  class Cipher
+    class V2
+      attr_reader :key, :iv, :cipher
+
+      def initialize(opts = {})
+        @cipher = OpenSSL::Cipher.new("aes-256-cbc")
+        @key    = opts[:key] ? Base64.strict_decode64(opts[:key]) : cipher.random_key
+        @iv     = opts[:iv] ? Base64.strict_decode64(opts[:iv]) : cipher.random_iv
+      end
+
+      def encrypt(plaintext)
+        c = cipher
+        c.encrypt
+        c.key = key
```

Comment:
@marcparadise This didn't feel like a particular blocker to testing. To move it to the initializer we would also need to pass in whether we are going to be used for encryption or decryption because you can't call these functions until after you call either encrypt or decrypt.